web/turkey/survey/views.py

```diff
@@ -1,106 +1,215 @@
-        if task_interaction_model.completed:
+
+class StepSubmission(RecordSubmission):
+    def process_submission(self, submission_data, task_interaction):
+        if task_interaction.task.external:
+            raise ValidationError(_('Submitting steps on an internal HIT'))
+
+        if self.associated_models_have_data(NAME_TO_STEP, task_interaction):
+            raise ValidationError(_('Step data already submitted for this task'))
+
+        self.save_data_to_mapped_models(submission_data['steps'],
+                                        NAME_TO_STEP,
+                                        task_interaction)
+
+
+class AuditorSubmission(RecordSubmission):
+    def process_submission(self, submission_data, task_interaction):
+        if self.associated_models_have_data(NAME_TO_AUDITOR, task_interaction):
+            raise ValidationError(_('Auditor data already submitted for this task'))
+
+        self.save_data_to_mapped_models(submission_data['auditors'],
+                                        NAME_TO_AUDITOR,
+                                        task_interaction)
+
+
+class CreateTaskInteractionView(APIView):
+    authentication_classes = (NonAuthentication,)
+    permission_classes = (AllowAny,)
+
+    def post(self, request, **kwargs):
+        if type(request.data) is not dict or 'task_pk' not in request.data:
             return Response(status=status.HTTP_400_BAD_REQUEST)
         try:
-            # if we fail validation or anything else all is well and nothing
-            # is committed
-            with transaction.atomic():
-                self.save_data_to_mapped_models(submission['auditors'],
-                                                NAME_TO_AUDITOR,
-                                                task_interaction_model)
-                self.save_data_to_mapped_models(submission['steps'],
-                                                NAME_TO_STEP,
-                                                task_interaction_model)
-                task_interaction_model.completed = True
-                task_interaction_model.save()
-            return Response(status=status.HTTP_201_CREATED)
-        except ValidationError:
+            task = Task.objects.get(pk=int(request.data['task_pk']))
+        except Task.DoesNotExist:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        except ValueError:
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
+        task_interaction = TaskInteraction.objects.create(task=task)
+        return Response(data={'task_interaction': task_interaction.pk,
+                              'auditor_submission_url': request.build_absolute_uri(
+                                  reverse('survey:auditor_submission',
+                                          kwargs={'pk': task_interaction.pk}))},
+                        status=status.HTTP_201_CREATED)
+
 
-@method_decorator(ensure_csrf_cookie, name='dispatch')
 class TaskView(View):
```

Comment:
Not all that different really, just adjusted to use an auditor embed template that powers both admin instructions for embedding and our own website. In other words we're dogfooding and it should keep us from neglecting either side of the project in regards to auditors.